salt_resources/states/basebox/miniconda.sls

```diff
@@ -0,0 +1,22 @@
+{% set current_path = salt['environ.get']('PATH', '/bin:/usr/bin') %}
+{% set miniconda_path = '/usr/local/bin/miniconda/bin' %}
+
+download_miniconda_installer:
+  cmd.run:
+    - name: wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O /home/{{ grains['deescalated_user'] }}/miniconda.sh
+    - unless: test -f {{ miniconda_path }}/conda
+    - runas: {{ grains['deescalated_user'] }}
+
+ensure_miniconda_is_installed:
+  cmd.run:
+    - name: bash /home/{{ grains['deescalated_user'] }}/miniconda.sh -b -p /usr/local/bin/miniconda
+    - unless: test -f {{ miniconda_path }}/conda
+    - runas: {{ grains['deescalated_user'] }}
+
+{% if not salt['file.search']('/home/{{ grains['deescalated_user'] }}/.bashrc', 'export PATH={{ miniconda_path }}:$PATH') %}
```

Comment:
Apparently Salt already checks for the string's existence in a file before appending it, so this `if` is unnecessary. https://docs.saltstack.com/en/latest/ref/states/all/salt.states.file.html#salt.states.file.append